docs/concepts/submit-data.md

```diff
@@ -58,6 +58,34 @@ The `GasCostPerBlobByte` and `GasCostPerTransactionByte` are parameters that
 could potentially be adjusted through the system's governance mechanisms. Hence,
 actual costs may vary depending on the current state of these parameters.
 
+#### Gas fee calculation
+
+The total fee for a transaction is calculated as the product of the gas limit
+for the transaction and the gas price set by the user:
+
+$\text{Total Fee} = \text{Gas Limit} \times \text{Gas Price}$
+
+The gas limit for a transaction is the maximum amount of gas that a user is
+willing to spend on a transaction. It is determined by both a static
+"fixed cost" and a dynamic cost based on the size of each blob involved
+in the transaction:
+
+$\text{Gas Limit} = \text{Fixed Cost} + \sum_{i=1}^{n} \text{Size of Blob}_i \times \text{Gas Cost Per Blob Byte}$
+
+Where:
+
+* $\text{Fixed Cost}$ is a static value (default is 65,000 gas)
+* $\sum_{i=1}^{n} \text{Size of Blob}_i$ is the size of the ith blob in the transaction
+* $\text{Gas Cost Per Blob Byte}$ is a parameter that could potentially be adjusted through the system's
+governance mechanisms.
+
+The gas price is set by the user when they submit a transaction. It is often
+specified by users directly. The total fee for the transaction is then
+calculated as the product of the estimated gas limit and the gas price.
+Since the state machine does not refund users for unused gas,
+it's important for users to estimate the gas limit accurately to
+avoid overpaying.
```

Comment:
Isn't it the case that users set the fee, not the gas price? Or did we change that?

Reply:
yep, thanks for catching that.